src/parquet/column/reader.h

```diff
@@ -152,6 +184,32 @@ class TypedColumnReader : public ColumnReader {
   std::vector<uint8_t> values_buffer_;
 };
 
+
+template <int TYPE>
+inline size_t TypedColumnReader<TYPE>::ReadValues(size_t batch_size, T* out) {
+  size_t num_decoded = current_decoder_->Decode(out, batch_size);
+  num_decoded_values_ += num_decoded;
+  return num_decoded;
+}
+
+template <int TYPE>
+inline size_t TypedColumnReader<TYPE>::ReadBatch(int batch_size, int16_t* def_levels,
+    int16_t* rep_levels, T* values, size_t* values_read) {
```

Comment:
Thanks -- I did some digging to clarify that `num_buffered_values_` is equal to the number of encoded levels (when the data is optional and/or repeated) and added a comment to that effect.